vortex-metrics/src/lib.rs

```diff
@@ -144,3 +168,120 @@ impl<'a> Iterator for VortexMetricsIter<'a> {
         self.iter.size_hint()
     }
 }
+
+impl DefaultTags {
+    fn merge(&self, other: &Self) -> Self {
+        DefaultTags(
+            self.0
+                .iter()
+                .chain(other.0.iter())
+                .map(|(k, v)| (k.clone(), v.clone()))
+                .collect(),
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_default_tags() -> Result<(), &'static str> {
+        let tags = [("file", "a"), ("partition", "1")];
+        let metrics = VortexMetrics::new_with_tags(tags.into_iter());
+
+        // Create a metric to verify tags
+        let counter = metrics.counter("test.counter");
+        counter.inc();
+        let snapshot = metrics.metrics();
+        let (name, metric) = snapshot.iter().next().unwrap();
+        assert_eq!(
+            name,
+            MetricId::new("test.counter")
+                .with_tag("file", "a")
+                .with_tag("partition", "1")
+        );
+        match metric {
+            Metric::Counter(c) => assert_eq!(c.count(), 1),
+            _ => return Err("metric is not a counter"),
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_multiple_children_with_different_tags() -> Result<(), &'static str> {
+        let parent_tags = [("service", "vortex")];
+        let parent = VortexMetrics::new_with_tags(parent_tags.into_iter());
+
+        let child1_tags = [("instance", "child1")];
+        let child2_tags = [("instance", "child2")];
+
+        let child1 = parent.child_with_tags(child1_tags.into_iter());
```

Comment:
nit - I think that if you change the `From` trait bound for `DefaultTags` you can drop these `into_iter`